docs/diagrams/MainLogicSequenceDiagram.puml

```diff
@@ -0,0 +1,52 @@
+@startuml MainLogicSequenceDiagram
+!include style.puml
+
+box "FridgeFriend\n" FRIDGEFRIEND_COLOR
+Participant ":FridgeFriend" as fridgefriend FRIDGEFRIEND_COLOR
+end box
+
+box Utilities UTILITIES_COLOR_T1
+Participant ":Ui" as ui UTILITIES_COLOR
+Participant ":Parser" as parser UTILITIES_COLOR
+end box
+
+box Command COMMAND_COLOR_T1
+Participant "command:Command" as command COMMAND_COLOR
+end box
+
+box Exception EXCEPTION_COLOR_T1
+Participant "e:Exception" as exception EXCEPTION_COLOR
+end box
+
+autoactivate on
+fridgefriend->fridgefriend: run()
+loop isExit
+    fridgefriend -> ui: getNextLine()
+    return userInput
+
+    alt successful execution
```

Comment:
Could it be possible that this is not an 'alt' box?

FridgeFriend runs the command regardless whether is an exception or not (unless invalid input, which can be mentioned, or perhaps simply put this as 'valid command' instead of 'successful execution')

So, I think if the 'alt's command is a successful execution, the logical opposite is an unsuccessful execution, in which case it affects the output but not the initial execution of the command.

"No more than one alternative partitions be executed in an alt frame. That is, it is acceptable for none of the alternative partitions to be executed but it is not acceptable for multiple partitions to be executed." (https://nus-cs2113-ay2021s2.github.io/website/schedule/week10/topics.html#w10-1-sequence-diagrams-basics)

Therefore I recommend either this condition be changed to "valid command" instead of "successful execution", or the alt box moved downward to capture execute() of Command rather than getCommand() of Parser.

Reply:
will edit based on prof's suggestions